### PR TITLE
Fix many flaky tests on macOs

### LINKED ISSUE #277: test: flaky Example_simpleQueueCustomMsgPack on macOS
https://github.com/tarantool/go-tarantool/actions/runs/4511463511/jobs/7943640255
```
2023-03-24T13:22:56.8314020Z === RUN   Example_simpleQueueCustomMsgPack
2023-03-24T13:22:59.5486520Z 2023/03/24 13:22:59 Take with timeout failed: client timeout for request 22 (0x4003)
2023-03-24T13:22:59.5539310Z FAIL	github.com/tarantool/go-tarantool/queue	33.950s
```

### LINKED ISSUE #278: test: flaky queue/Example_connectionPool on macOS
https://github.com/tarantool/go-tarantool/actions/runs/4511463511/jobs/7944401745
```
2023-03-24T13:52:48.0576220Z === RUN   Example_connectionPool
2023-03-24T13:52:49.1545340Z 2023/03/24 13:52:49 tarantool: storing connection to 127.0.0.1:3015 canceled: .../.rocks/share/tarantool/queue/abstract/queue_session.lua:223: The UUID f16c5fac-9d33-4640-8494-1043d26b4b80 is unknown. (0x20)
2023-03-24T13:52:49.1556050Z --- FAIL: Example_connectionPool (1.10s)
```

https://github.com/tarantool/go-tarantool/actions/runs/4512321687/jobs/7948237951
```
--- FAIL: Example_connectionPool (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x438ed37]

goroutine 1 [running]:
testing.(*InternalExample).processRunResult(0xc0001e95a8, 0xc0000289f0, 0x28, 0xcbff1f, 0x4438b40, 0x4825660, 0x4015a70)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/testing/example.go:89 +0x63f
testing.runExample.func2(0xc0ff93057598e2e0, 0x6ec8ac3cc, 0x4831f20, 0xc0001184d0, 0xc0000aa008, 0xc000131440, 0xc0001e95a8, 0xc0001e95d8)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/testing/run_example.go:58 +0x105
panic(0x4438b40, 0x4825660)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/runtime/panic.go:679 +0x1b2
github.com/tarantool/go-tarantool.(*Connection).loadSchema(0xc000296000, 0x44d11b8, 0xc000296000)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/schema.go:328 +0x397
github.com/tarantool/go-tarantool.Connect(0x44b6cd8, 0xe, 0x0, 0x0, 0x0, 0x3b9aca00, 0x0, 0x0, 0x44b3ee0, 0x4, ...)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/connection.go:430 +0x65d
github.com/tarantool/go-tarantool/connection_pool.(*ConnectionPool).fillPools(0xc000098b60, 0xc00010ebd0, 0x44b6cd8, 0xe, 0xc00010ec89)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/connection_pool/connection_pool.go:905 +0x1fd
github.com/tarantool/go-tarantool/connection_pool.ConnectWithOpts(0xc00010f2d0, 0x2, 0x2, 0x0, 0x0, 0x0, 0x3b9aca00, 0x0, 0x0, 0x44b3ee0, ...)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/connection_pool/connection_pool.go:146 +0x5fb
github.com/tarantool/go-tarantool/queue_test.Example_connectionPool()
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/queue/example_connection_pool_test.go:151 +0x2d9
testing.runExample(0x44ba4e8, 0x16, 0x44d1208, 0x44cb6ba, 0x9a, 0x0, 0x0)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/testing/run_example.go:62 +0x209
testing.runExamples(0xc0001e9770, 0x482b820, 0x3, 0x3, 0x101)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/testing/example.go:44 +0x1a8
testing.(*M).Run(0xc00012a200, 0x0)
	/Users/runner/hostedtoolcache/go/1.13.15/x64/src/testing/testing.go:1118 +0x203
github.com/tarantool/go-tarantool/queue_test.runTestMain(0xc00012a200, 0x0)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/queue/queue_test.go:952 +0x530
github.com/tarantool/go-tarantool/queue_test.TestMain(0xc00012a200)
	/Users/runner/work/go-tarantool/go-tarantool/tarantool/go-tarantool/queue/queue_test.go:956 +0x2b
main.main()
	_testmain.go:84 +0x135
FAIL	github.com/tarantool/go-tarantool/queue	30.049s
```

### LINKED ISSUE #282: Flaky macOs tests
Faced some more flaky tests for macOs on github runners.
I'll update the list if I find any other failed tests.
 See the list below:
1.  [run](https://github.com/tarantool/go-tarantool/actions/runs/4607708178/jobs/8143198363)
```
--- FAIL: ExampleConnection_Update (0.01s)
got:
Update 14
Error <nil>
Code 0
Data [[14 bye one]]
want:
Update 14
Error <nil>
Code 0
Data [[14 bye bla]]
```
```
--- FAIL: ExampleConnection_SelectAsync (0.01s)
got:
Future 0 Error <nil>
Future 0 Data [{{} 16 test one} {{} 1 hello world}]
Future 1 Error <nil>
Future 1 Data [[17 test one]]
Future 2 Error <nil>
Future 2 Data [[18 test one]]
want:
Future 0 Error <nil>
Future 0 Data [{{} 16 val 16 bla} {{} 15 val 15 bla}]
Future 1 Error <nil>
Future 1 Data [[17 val 17 bla]]
Future 2 Error <nil>
Future 2 Data [[18 val 18 bla]]
```
```
--- FAIL: TestClient (0.51s)
    tarantool_test.go:886: Failed to Insert: client timeout for request 8 (0x4003)
```
2. [run](url)
```
2023/04/04 13:37:51 Failed to prepare test tarantool: failed to dial: dial tcp 127.0.0.1:3013: connect: connection refused
FAIL	github.com/tarantool/go-tarantool/uuid	5.703s
```
3. [run](url)
```
--- FAIL: TestTtlQueue (12.61s)
    queue_test.go:695: Failed take from queue: read tcp 127.0.0.1:50792->127.0.0.1:3013: i/o timeout
    queue_test.go:699: Failed to drop queue: using closed connection (0x4001)
```
```
--- FAIL: TestTask_Touch (1.97s)
    --- FAIL: TestTask_Touch/test_queue (0.21s)
        queue_test.go:869: Failed to create queue: ...queue/testdata/.rocks/share/tarantool/queue/abstract.lua:597: Space 'test_queue' already exists (LuajitError, code 0x20), see ./src/lua/utils.c line 489
```
```
--- FAIL: Example_simpleQueue (1.77s)
got:
error in prepare is failed to dial: dial tcp 127.0.0.1:3013: i/o timeout
want:
data_1:  test_data_1
FAIL
```

```
--- FAIL: TestTtlQueue_Put (0.38s)
    queue_test.go:711: Failed to create queue: ...queue/testdata/.rocks/share/tarantool/queue/abstract.lua:597: Space 'test_queue' already exists (LuajitError, code 0x20), see ./src/lua/utils.c line 489
```
4. [run](https://github.com/tarantool/go-tarantool/actions/runs/4606670484/jobs/8141511209)
```
--- FAIL: TestInsert (0.66s)
    connection_pool_test.go:1196: 
        	Error Trace:	connection_pool_test.go:1196
        	Error:      	Expected nil, but got: tarantool.ClientError{Code:0x4003, Msg:"client timeout for request 10"}
        	Test:       	TestInsert
        	Messages:   	failed to Insert
```

### LINKED ISSUE #288: test: flaky crud/TestCrudGenerateData
```
2023-04-25T19:23:49.4163123Z --- FAIL: TestCrudGenerateData (0.01s)
2023-04-25T19:23:49.4163662Z     --- PASS: TestCrudGenerateData/Insert (0.00s)
2023-04-25T19:23:49.4164227Z     --- FAIL: TestCrudGenerateData/InsertObject (0.00s)
2023-04-25T19:23:49.4167665Z         tarantool_test.go:467: Failed to perform CRUD request on CRUD side: map[interface {}]interface {}{"class_name":"InsertError", "err":"Failed to insert: Duplicate key exists in unique index \"primary_index\" in space \"test\" with old tuple - [1019, 222, \"bla\"] and new tuple - [1019, 222, \"bla\"]", "file":"...ool/crud/testdata/.rocks/share/tarantool/crud/insert.lua", "line":0x70, "str":"InsertError: Failed to insert: Duplicate key exists in unique index \"primary_index\" in space \"test\" with old tuple - [1019, 222, \"bla\"] and new tuple - [1019, 222, \"bla\"]"}
```
[log.txt](https://github.com/tarantool/go-tarantool/files/11326246/log.txt)

### LINKED ISSUE #291: test: flaky queue tests initialization
We have a problem to start Tarantool instances on macOS:

```
2023-05-14T11:55:28.6833270Z PASS
2023-05-14T11:55:29.7810430Z ok  	github.com/tarantool/go-tarantool/multi	14.836s
2023-05-14T11:55:31.3979520Z # github.com/tarantool/go-tarantool/queue.test
2023-05-14T11:55:31.3980060Z ld: warning: -no_pie is deprecated when targeting new OS versions
2023-05-14T11:55:45.2246100Z 2023/05/14 11:55:44 Failed to prepare test tarantool: failed to dial: dial tcp 127.0.0.1:3013: connect: connection refused
2023-05-14T11:55:45.3231560Z FAIL	github.com/tarantool/go-tarantool/queue	9.036s
2023-05-14T11:55:50.6396720Z # github.com/tarantool/go-tarantool/settings.test
2023-05-14T11:55:50.6397790Z ld: warning: -no_pie is deprecated when targeting new OS versions
2023-05-14T11:55:50.9457490Z 2023/05/14 11:55:50 Skipping session settings tests...
2023-05-14T11:55:50.9463290Z === RUN   TestRequestsAPI
```

```
2023-05-14T12:03:03.9143630Z PASS
2023-05-14T12:03:04.9534080Z ok  	github.com/tarantool/go-tarantool/multi	14.695s
2023-05-14T12:03:06.4570850Z # github.com/tarantool/go-tarantool/queue.test
2023-05-14T12:03:06.4571360Z ld: warning: -no_pie is deprecated when targeting new OS versions
2023-05-14T12:03:13.1736710Z 2023/05/14 12:03:13 Failed to set roles in tarantool pool: failed to authenticate: Instance bootstrap hasn't finished yet (ClientError, code 0x74), see ./src/box/box.cc line 3037
2023-05-14T12:03:13.1756560Z FAIL	github.com/tarantool/go-tarantool/queue	6.627s
2023-05-14T12:03:14.6679980Z # github.com/tarantool/go-tarantool/settings.test
2023-05-14T12:03:14.6681190Z ld: warning: -no_pie is deprecated when targeting new OS versions
2023-05-14T12:03:15.1690520Z === RUN   TestRequestsAPI
2023-05-14T12:03:15.1699400Z --- PASS: TestRequestsAPI (0.00s)
```

### PR DESCRIPTION
**Pull request by oleg-jukovec**
Fixed many flaky tests for macOs.

I didn't forget about (remove if it is not applicable):

- [x] Tests (see [documentation](https://pkg.go.dev/testing) for a testing package)
- [x] Changelog (see [documentation](https://keepachangelog.com/en/1.0.0/) for changelog format)

Related issues:

Closes #277
Closes #278 
Closes #282
Closes #288
Closes #291

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ clean:
 .PHONY: deps
 deps: clean
 	( cd ./queue/testdata; $(TTCTL) rocks install queue 1.3.0 )
-	( cd ./crud/testdata; $(TTCTL) rocks install crud 1.0.0 )
+	( cd ./crud/testdata; $(TTCTL) rocks install crud 1.1.1 )
```
Comment (better0fdead) [inline Makefile:30]: Mb add a changelog entry about it.

**Comment by better0fdead** [comment]
[test: wait for queue cluster](https://github.com/tarantool/go-tarantool/pull/292/commits/a21510843dbdc527aa4e3aff9df1561aa3b8329b) 
We need to wait for a successful role cluster configuration because an instance bootstrap may haven't been finished yet.

File: Makefile (hunk 1 above)
Comment (oleg-jukovec) [reply Makefile:30]: We don't have a release since crud support is added. So we don't need a record in the changelog.

**Comment by oleg-jukovec** [comment]
> [test: wait for queue cluster](https://github.com/tarantool/go-tarantool/pull/292/commits/a21510843dbdc527aa4e3aff9df1561aa3b8329b) We need to wait for a successful role cluster configuration because an instance bootstrap may haven't been finished yet.

Thank you, updated.

**Review by better0fdead — APPROVED** [review]

**Review by vr009 — APPROVED** [review]

**Pull request merged**